.github/workflows/build.yml

```diff
@@ -122,12 +122,40 @@ jobs:
         run: cargo build --workspace --all-targets --all-features ${{ matrix.config.extraArgs }}
 
       - name: Cargo Test
+        if: ${{ fromJSON(matrix.config.os != 'ubuntu-latest') }}
         run: |
           make test-unit
           make test-e2e
         env:
           RUST_LOG: spin=trace
 
+      - name: Install cargo-tarpaulin binary crate (ubuntu)
+        uses: actions-rs/install@v0.1
+        if: ${{ fromJSON(matrix.config.os == 'ubuntu-latest') }}
+        with:
+          crate: cargo-tarpaulin
+          version: latest
+          use-tool-cache: true
+
+      - name: Cargo Tarpaulin (ubuntu)
+        if: ${{ fromJSON(matrix.config.os == 'ubuntu-latest') }}
+        run: cargo tarpaulin --follow-exec --skip-clean -t 6000 --out xml --features openssl/vendored,default,e2e-tests
+        env:
+          RUST_LOG: spin=trace
+
+      - name: Upload to codecov.io (ubuntu)
+        uses: codecov/codecov-action@v1.0.2
```

Comment:
While reading up on this action, I noticed the following disclaimer: https://github.com/codecov/codecov-action#%EF%B8%8F--deprecation-of-v1

Can we bump to a more recent (3.x?) version?

Reply:
Updated it. Thanks!